src/Repository/PostRepository.php

```diff
@@ -24,35 +24,25 @@
  *
  * See https://symfony.com/doc/current/doctrine.html#querying-for-objects-the-repository
  *
- * @author Ryan Weaver <weaverryan@gmail.com>
- * @author Javier Eguiluz <javier.eguiluz@gmail.com>
- * @author Yonel Ceruto <yonelceruto@gmail.com>
- *
  * @method Post|null findOneByTitle(string $postTitle)
  *
- * @template-extends ServiceEntityRepository<Post>
+ * @template-extends ServiceDocumentRepository<Post>
  */
-class PostRepository extends ServiceEntityRepository
+class PostRepository extends ServiceDocumentRepository
 {
     public function __construct(ManagerRegistry $registry)
     {
         parent::__construct($registry, Post::class);
     }
 
-    public function findLatest(int $page = 1, Tag $tag = null): Paginator
+    public function findLatest(int $page = 1, ?string $tag = null): Paginator
     {
-        $qb = $this->createQueryBuilder('p')
-            ->addSelect('a', 't')
-            ->innerJoin('p.author', 'a')
-            ->leftJoin('p.tags', 't')
-            ->where('p.publishedAt <= :now')
-            ->orderBy('p.publishedAt', 'DESC')
-            ->setParameter('now', new \DateTime())
-        ;
+        $qb = $this->createQueryBuilder()
+            ->field('publishedAt')->lte(new \DateTimeImmutable()) // $$NOW is a MongoDB variable that returns the current date
```

Comment:
You can use reference priming to eagerly load all referenced authors:

```suggestion
            ->field('publishedAt')->lte(new \DateTimeImmutable()) // $$NOW is a MongoDB variable that returns the current date
            ->field('author')->prime(true)
```

Reply:
It looks like primers are only supported for `TYPE_FIND_AND_UPDATE` and `TYPE_FIND_AND_REMOVE` queries. This is a `TYPE_FIND` query :grimacing:
https://github.com/doctrine/mongodb-odm/blob/2.5.x/lib/Doctrine/ODM/MongoDB/Query/Query.php#L213-L231

Reply:
For queries of `TYPE_FIND`, this is done in `makeIterator`, which is called from `runQuery`: https://github.com/doctrine/mongodb-odm/blob/2.5.x/lib/Doctrine/ODM/MongoDB/Query/Query.php#L399..L402